include/robot_interfaces/robot_logger.hpp

```diff
@@ -0,0 +1,331 @@
+        : block_size_(block_size),
+          stop_was_called_(false),
+          logger_data_(robot_data)
+    {
+        thread_ = std::make_shared<real_time_tools::RealTimeThread>();
+    }
+
+    virtual ~RobotLogger()
+    {
+        stop_was_called_ = true;
+        stop();
+        thread_->join();
+    }
+
+    std::vector<std::string> get_header()
+    {
+
+        std::vector<std::string> observation_name = observation.get_name();
+        std::vector<std::string> applied_action_name =
+            applied_action.get_name();
+        std::vector<std::string> desired_action_name =
+            desired_action.get_name();
+        std::vector<std::string> status_name = status.get_name();
+
+        std::vector<std::vector<double>> observation_data =
+            observation.get_data();
+        std::vector<std::vector<double>> applied_action_data =
+            applied_action.get_data();
+        std::vector<std::vector<double>> desired_action_data =
+            desired_action.get_data();
+        std::vector<std::vector<double>> status_data = status.get_data();
+
+        // using the newest index information in logger_data to extract
+        // information on the fields to be logged
+        applied_action =
+            (*logger_data_->applied_action)[logger_data_->applied_action
+                                                ->newest_timeindex()];
+        desired_action =
+            (*logger_data_->desired_action)[logger_data_->desired_action
+                                                ->newest_timeindex()];
+        observation =
+            (*logger_data_
+                  ->observation)[logger_data_->observation->newest_timeindex()];
+        status =
+            (*logger_data_->status)[logger_data_->status->newest_timeindex()];
+
+        header_.push_back("#");
+        header_.push_back("[Timestamp]");
+        header_.push_back("(Time Index)");
+
+        append_name_to_header(status_name, status_data);
+        append_name_to_header(observation_name, observation_data);
+        append_name_to_header(applied_action_name, applied_action_data);
+        append_name_to_header(desired_action_name, desired_action_data);
+
+        return header_;
+
+    }
+
+    void append_name_to_header(std::vector<std::string> field_name,
+                               std::vector<std::vector<double>> field_data)
+    {
+
+        int i, j;
+        std::string temp;
+
+        for (i = 0; i < field_name.size(); i++)
+        {
+          if(field_data[i].size() == 1)
+          {
+            temp = field_name[i];
+            header_.push_back(temp);
+          }
+          else
+          {
+            for (j = 0; j < field_data[i].size(); j++)
+            {
+                temp = field_name[i] + " " + std::to_string(j);
+                header_.push_back(temp);
+            }
+          }
+        }
+
+    }
+
+    void append_header_to_file()
+    {
+
+      output_file_.open(output_file_name_, std::ios_base::app);
```

Comment:
Indentation does not follow our style guide.

Reply:
Yeah, I forgot to check again, had not cherry-picked the format commit. Sorry, fixed.